Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift

```diff
@@ -235,7 +235,7 @@ extension BinarySearchTree {
   /*
     Finds the node whose value precedes our value in sorted order.
   */
-  public func predecessor() -> BinarySearchTree<T>? {
+  public func predecessor() -> BinarySearchTree<T> {
```

Comment:
Taking away the optionality of the return type isn't going to work - we need to be able to return `nil` if the current node has no predecessor.

Reply:
Yes. You're right. I send new PR to correct. PTAL! 👍